src-cpp/app/modules/main/wallet/accounts/view.h

```diff
@@ -5,30 +5,46 @@
 #include <memory>
 
 #include "model.h"
+#include "controller.h"
+#include "item.h"
 
 namespace Modules::Main::Wallet::Accounts
 {
 class View : public QObject
 {
     Q_OBJECT
     Q_PROPERTY(Model* model READ getModel NOTIFY modelChanged)
+    Q_PROPERTY(Item* currentAccount READ getCurrentAccount NOTIFY currentAccountChanged)
 
 public:
-    explicit View(QObject* parent = nullptr);
+    explicit View(Controller* controller, QObject* parent = nullptr);
     ~View() = default;
 
     void load();
-    void setModelItems(QVector<Item> &accounts);
+    void setModelItems(QVector<Item*> accounts);
+
+    Q_INVOKABLE QString generateNewAccount(QString password, QString accountName, QString color);
+    Q_INVOKABLE QString addAccountsFromPrivateKey(QString privateKey, QString password, QString accountName, QString color);
+    Q_INVOKABLE QString addAccountsFromSeed(QString seedPhrase, QString password, QString accountName, QString color);
+    Q_INVOKABLE QString addWatchOnlyAccount(QString address, QString accountName , QString color);
+    Q_INVOKABLE void deleteAccount(QString address);
+    Q_INVOKABLE void switchAccount(int index);
 
 private:
+    void refreshWalletAccounts();
+
     Model* m_modelPtr;
+    Controller* m_controllerPtr;
+    Item* m_currentAccountPtr;
 
 public slots:
     Model* getModel();
+    Item* getCurrentAccount();
```

Comment:
why slots return something? if you call it by `connection` its useless, but if you use it in QML - it should be just `Q_INVOKABLE`

Reply:
> I found a several topics which we can improve here. The best way to keep in mind it:
> 
> * Every `complex` (not basic types) input parameter should be  passed as `const Type&`
> * Every `complex` return type can be `const Type&` (except temporary objects which can be moved)
> * Every `getter` should be a const.
> * `nullptr` convertible to `bool` safety
> * Every pointer as input parameters should be check for `nullptr`
> * Every non-void function must return something anyway (can be fixed by compiler parameter )
> * Avoid unnecessary type conversions (for example, from double to int).
> * Add default parameters to Constructors.
> * Try to use build-in methods like `isEmpty` or `isValid`.
> * Check `out-of-range` for containers.
> 
> This is a `basic` and `necessary` requirements to efficient and stable C++ code. We can skip some `super-new-standard-features`, but that list is `must be`.

should create a cpp code checklist